<?xml version="1.0"?>

<robot name="multipleshapes">

  <!-- #region MATERIAL-->
  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <!-- #endregion -->

  <!-- #region BASE_LINK -->
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.6" radius="0.2"/>
      </geometry>
      <material name="white"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.6" radius="0.2"/>
      </geometry>
    </collision>
    <inertia>
      <mass value="10"/>
      <inertia ixx="0.4" ixy="0.0" ixz="0.0" iyy="0.4" iyz="0.0" izz="0.2"/>
    </inertia>
  </link>
  <!-- #endregion -->

  <!-- #region RIGHT_LEG -->
  <link name="right_leg">
    <visual>
      <geometry>
        <box size="0.6 0.1 0.2"/>
      </geometry>
      <!-- move visual origin away from true origin by xyz -->
      <!-- rotate by rpy around xyz axis -->
      <origin rpy="0 1.57075 0" xyz="0 0 -0.3"/>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="base_to_right_leg" type="revolute">
    <parent link="base_link"/>
    <child link="right_leg"/>
    <!-- joint of child is xyz away from parent -->
    <origin xyz="0 -0.22 0.25"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-1" upper="1" velocity="0.5"/>
  </joint>
  <!-- #endregion -->

  <!-- #region LEFT_LEG -->
  <link name="left_leg">
    <visual>
      <geometry>
        <box size="0.6 0.1 0.2"/>
      </geometry>
      <!-- move visual origin away from true origin by xyz -->
      <!-- rotate by rpy around xyz axis -->
      <origin rpy="0 1.57075 0" xyz="0 0 -0.3"/>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="base_to_left_leg" type="revolute">
    <parent link="base_link"/>
    <child link="left_leg"/>
    <!-- joint of child is xyz away from parent -->
    <origin xyz="0 0.22 0.25"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-1" upper="1" velocity="0.5"/>
  </joint>
  <!-- #endregion -->

  <!-- #region HEAD -->
  <link name="head">
    <visual>
      <geometry>
        <sphere radius="0.2"/>
      </geometry>
      <!-- move visual origin away from true origin by xyz -->
      <!-- rotate by rpy around xyz axis -->
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="base_to_head" type="fixed">
    <parent link="base_link"/>
    <child link="head"/>
    <!-- joint of child is xyz away from parent -->
    <origin xyz="0 0 0.3"/>
  </joint>
  <!-- #endregion -->

  <!-- #region LEFT_WHEEL -->
  <link name="left_wheel">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <!-- move visual origin away from true origin by xyz -->
      <!-- rotate by rpy around xyz axis -->
      <origin rpy="1.57075 0 0" xyz="0 0 0"/>
      <material name="white"/>
    </visual>
  </link>

  <joint name="left_leg_to_left_wheel" type="continuous">
    <parent link="left_leg"/>
    <child link="left_wheel"/>
    <!-- joint of child is xyz away from parent -->
    <origin xyz="0 0 -0.6"/>
    <axis xyz="0 1 0"/>
  </joint>
  <!-- #endregion -->
  
  <!-- #region RIGHT_WHEEL -->
  <link name="right_wheel">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.1"/>
      </geometry>
      <!-- move visual origin away from true origin by xyz -->
      <!-- rotate by rpy around xyz axis -->
      <origin rpy="1.57075 0 0" xyz="0 0 0"/>
      <material name="white"/>
    </visual>
  </link>

  <joint name="right_leg_to_right_wheel" type="continuous">
    <parent link="right_leg"/>
    <child link="right_wheel"/>
    <!-- joint of child is xyz away from parent -->
    <origin xyz="0 0 -0.6"/>
    <axis xyz="0 1 0"/>
  </joint>
  <!-- #endregion -->
</robot>